test/upgrade/test-upgrade.bats

```diff
@@ -250,7 +255,26 @@ failed    | exited     | 17
 
 @test "network - restart" {
     # restart the container and check if we can still use the port
+
+    # https://github.com/containers/podman/issues/13679
+    # The upgrade to podman4 changes the network db format.
+    # While it is compatible from 3.X to 4.0 it will fail the other way around.
+    # This can be the case when the cleanup process runs before the stop process
+    # can do the cleanup.
+
+    # Since there is no easy way to fix this and downgrading is not something
+    # we support, just fix this bug in the tests by manually calling
+    # network disconnect to teardown the netns.
+    if NETWORK_CONNECT_TEST; then
```

Comment:
...then here you do `if test -e $PODMAN_UPGRADE_WORKDIR/ran-network-connect-test; then`